File: backend/src/main/java/com/wecp/progressive/entity/Student.java

package com.wecp.progressive.entity;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "student")
public class Student implements Comparable<Student>{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "student_id")
    private Integer studentId;

    @Column(name = "full_name")    
    private String fullName;

    @Column(name = "date_of_birth") 
    private Date dateOfBirth;

    @Column(name = "contact_number") 
    private String contactNumber;

    @Column(name = "email") 
    private String email;
    
    @Column(name = "address") 
    private String address;

    public Student() {
    }

    public Student(Integer studentId, String fullName, Date dateOfBirth, String contactNumber, String email,
            String address) {
        this.studentId = studentId;
        this.fullName = fullName;
        this.dateOfBirth = dateOfBirth;
        this.contactNumber = contactNumber;
        this.email = email;
        this.address = address;
    }

    public Integer getStudentId() {
        return studentId;
    }

    public void setStudentId(Integer studentId) {
        this.studentId = studentId;
    }

    public String getFullName() {
        return fullName;
    }

    public void setFullName(String fullName) {
        this.fullName = fullName;
    }

    public Date getDateOfBirth() {
        return dateOfBirth;
    }

    public void setDateOfBirth(Date dateOfBirth) {
        this.dateOfBirth = dateOfBirth;
    }

    public String getContactNumber() {
        return contactNumber;
    }

    public void setContactNumber(String contactNumber) {
        this.contactNumber = contactNumber;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    @Override
    public int compareTo(Student o) {
        return this.fullName.compareTo(o.getFullName());
    }
}

File: frontend/src/app/educonnect/models/Student.ts
export class Student {
    studentId: number;
    fullName: string;
    dateOfBirth: Date | null;
    contactNumber: string;
    email: string;
    address: string;

    constructor(studentId: number,fullName: string, dateOfBirth: Date | null, contactNumber: string, email: string,
        address: string) {
        this.studentId = studentId;
        this.fullName = fullName;
        this.dateOfBirth = dateOfBirth;
        this.contactNumber = contactNumber;
        this.email = email;
        this.address = address;
    }

    logAttributes(): void {
        console.log('studentId:', this.studentId);
        console.log('fullName:', this.fullName);
        console.log('dateOfBirth:', this.dateOfBirth);
        console.log('contactNumber:', this.contactNumber);
        console.log('email:', this.email);
        console.log('address:', this.address);
    }
}